tests/stdlib/tdecode_helpers.nim

```diff
@@ -0,0 +1,27 @@
+import std/private/decode_helpers
+
+
+block:
```

Comment:
ok for this PR but in future I really prefer using the magic of templates to reduce boilerplate to 0:
```nim
template tdecodePercent(s): untyped =
  var i = 0
  let c = decodePercent(s, i)
  (i, c)
# then it's just 1 liners; furthermore it makes it easy if you want to edit
# the test as it's all in 1 place => DRY
doAssert tdecodePercent("%t9") == (0, '%')
doAssert tdecodePercent("19") == (0, '%')
...
```

(or even moving the doAssert inside, depending on your taste)

Reply:
will do it in the future PR.